### PR TITLE
fix: oracle chokepoint guard + effective_model helper

### PR DESCRIPTION
**Pull request by EYH0602**
## Summary

Follow-up to #173. The reviewer flagged that the fix landed in `src/benchflow/cli/eval.py`, but `bench eval create` actually dispatches to `src/benchflow/cli/main.py:707` — `cli/eval.py` is not wired into the live CLI. On top of that, #173 deliberately removed the chokepoint guard from `_agent_env.py:147` on the assumption that the CLI would pre-filter, so oracle is now *more* broken on the live code path: `bench eval create -a oracle` raises `ValueError("ANTHROPIC_API_KEY required for model 'claude-haiku-4-5-20251001' but not set …")` even though oracle never calls an LLM.

Three layers, one PR:

- **Layer 1 — restore the chokepoint** (`_agent_env.py:147`). Put back `if model and agent != "oracle":`. Single point of defense; catches every caller (current + future, CLI + SDK + YAML).
- **Layer 2 — clarify the not-yet-live module**. Keep `src/benchflow/cli/eval.py` and `tests/test_eval_cli.py` (future-facing eval design), but add docstring notes that they are not wired into the live CLI. Update `eval.py` to use the `effective_model()` helper for the oracle guard. A regression test ensures `cli/main.py` does not import `cli/eval`.
- **Layer 3 — kill the structural pressure**. New helper `effective_model(agent, model)` in `job.py`. Change `JobConfig.model` default to `None`. Replace all 7 `model or DEFAULT_MODEL` sites in `cli/main.py` and `job.py` YAML loaders with the helper. Oracle gets honest `model=None` end-to-end (in `result.json`, `summary.json`, prints).

## Regression tests

`tests/test_resolve_env_helpers.py::TestResolveAgentEnvOracle` pins the chokepoint at unit level. `tests/test_oracle_chokepoint.py` (13 tests) pins each layer at the right altitude:

| Class | What it pins |
|---|---|
| `TestEvalModuleNotWiredIntoCLI` | `cli/main.py` does not import `cli/eval` |
| `TestEvalCreateRouting` | `bench eval create` → `cli/main.py:eval_create` |
| `TestEffectiveModel` | helper behaviour for oracle / non-oracle / empty model |
| `TestOracleYamlLoaders` | `Job.from_yaml(oracle config) → model is None` (native + Harbor) |
| `TestEvalCreateOracleCLI` | end-to-end: live `eval_create(agent="oracle")` + no API key → no raise |

## Test plan

- [x] `pytest tests/test_oracle_chokepoint.py` → 13 passed
- [x] `pytest tests/test_eval_cli.py` → 7 passed
- [x] `pytest tests/test_resolve_env_helpers.py` → all passed
- [x] Full suite: same pre-existing failures as `main` (Docker not running locally). Zero new failures.
- [x] `ty check src/`: same pre-existing diagnostics as `main`. Zero new ones.

---

<a href="https://app.devin.ai/review/benchflow-ai/benchflow/pull/174" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open in Devin Review
  </picture>
</a>

### CODE REVIEW

**Review by EYH0602 — COMMENTED** [review]
## Review Summary
Reviewed 7 files, 346 lines added / 28 deleted across bug detection, type design, test coverage, and guidelines compliance.

## Critical Issues (confidence 90-100)
None.

## Important Issues (confidence 75-89)
None.

All findings scored below the 75 confidence threshold — no actionable issues found.

## Positive Observations

- **Three-layer defense is excellent architecture.** Layer 1 (chokepoint guard in `resolve_agent_env`), Layer 2 (docstring/structural clarification on the not-yet-live `cli/eval.py`), and Layer 3 (`effective_model()` helper) each defend independently — any single layer failing still leaves the others protecting.

- **`effective_model()` eliminates 9 duplicate `model or DEFAULT_MODEL` sites** across `cli/main.py`, `cli/eval.py`, and the YAML loaders in `job.py`. Centralizing the rule "oracle → None, else → model or default" in one helper is a meaningful duplication reduction.

- **Test layering matches the code layering.** Tests pin each defense at the right altitude: `TestResolveAgentEnvOracle` pins Layer 1, `TestEvalModuleNotWiredIntoCLI` pins Layer 2 structurally, `TestEffectiveModel` pins the helper, `TestOracleYamlLoaders` pins the YAML entry points, and `TestEvalCreateOracleCLI` pins the full CLI → chokepoint path end-to-end. 13 new tests for a 3-line core change is proportionate to the bug's subtlety.

- **Structural regression test is clever.** `test_cli_main_does_not_import_cli_eval` reads the source text of `cli/main.py` and asserts no `cli/eval` import — this directly prevents the mistake that made PR #173's fix land in dead code.

- **JobConfig.model default → None is clean.** Forcing callers through `effective_model()` means config objects carry honest data (oracle shows `model=null` in `summary.json`, not a bogus default). All internal callers were updated.

- **Edge cases covered.** Empty string model from Harbor YAML, explicit model forwarded to oracle, non-oracle fallback to DEFAULT_MODEL — all verified in tests.

Verified: all 9 `model or DEFAULT_MODEL` sites in `cli/main.py` + `cli/eval.py` are replaced, both YAML loaders in `job.py` route through the helper, and `DEFAULT_MODEL` is only imported where still needed. No remaining unguarded call sites. LGTM — request an independent reviewer per CLAUDE.md policy.

**Comment by EYH0602** [comment]
@xdotli this fix should do the work

**Comment by xdotli** [comment]
> @xdotli this fix should do the work

waiting for @devin to review it

File: src/benchflow/cli/eval.py [hunk 1]
```diff
@@ -1,7 +1,12 @@
 """`bf eval` — the benchflow eval-runner command group.
 
-The future-facing entry point for running evaluations. Anthropic-style shape:
-resource creation, one command, return the result or a job-id.
+NOTE: This module is **not wired into the live CLI**.  The active
```
Comment (xdotli) [inline src/benchflow/cli/eval.py:3]: @claude is this comment good?

**Review by xdotli — APPROVED** [review]
